### PR TITLE
Added Tests for Eclipse Plugin

### PR DESCRIPTION
**Pull request by tmess567**
I have yet to test out the Travis config to see if it can run these tests.
Also Editor Tests are not complete yet because for some reason swtbot is unable to pick up the resource bundle for the HTMLEditor.

These tests check if the admin login works and whether or not mail and report templates can be added and deleted using the tree view.

### CODE REVIEW

**Comment by ilgrosso** [comment]
Linking to SYNCOPE-809

File: ide/eclipse/bundles/org.apache.syncope.ide.eclipse.tests/pom.xml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ilgrosso) [inline ide/eclipse/bundles/org.apache.syncope.ide.eclipse.tests/pom.xml]: Why tests in a separate Maven module?

File: ide/eclipse/bundles/org.apache.syncope.ide.eclipse.tests/pom.xml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        <configuration>
+          <useUIHarness>true</useUIHarness>
+          <useUIThread>false</useUIThread>
+          <product>org.eclipse.sdk.ide</product>
+          <application>org.eclipse.ui.ide.workbench</application>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+
```
Comment (ilgrosso) [inline ide/eclipse/bundles/org.apache.syncope.ide.eclipse.tests/pom.xml]: Where is the `skipTests` profile?

File: ide/eclipse/bundles/org.apache.syncope.ide.eclipse.tests/src/main/java/org/apache/syncope/ide/eclipse/tests/SyncopeViewTestCase.java [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+    private SWTBotTreeItem mTemplateItem;
+    private SWTBotTreeItem rTemplateItem;
+    private SWTBotTreeItem[] mtemplateList;
+    private SWTBotTreeItem[] rtemplateList;
+
+    protected static final String ADMIN_UNAME = "admin";
+
+    protected static final String ADMIN_PWD = "password";
+
+    protected static final String ADDRESS = "http://localhost:9080/syncope/rest";
```
Comment (ilgrosso) [inline ide/eclipse/bundles/org.apache.syncope.ide.eclipse.tests/src/main/java/org/apache/syncope/ide/eclipse/tests/SyncopeViewTestCase.java]: Do such test cases require Apache Syncope core running? (I think so)
If so, you should refactor things to include your test cases under `fit/core-reference/src/test/java`

**Comment by ilgrosso** [comment]
Overall, as you can read from comments, I think it would be better to move the test cases under `fit/core-reference`.
Moreover, isn't there any way to run such tests headless, e.g. without requiring Xvfb and `DISPLAY`?

File: ide/eclipse/bundles/org.apache.syncope.ide.eclipse.tests/pom.xml (hunk 1 above)
Comment (tmess567) [inline ide/eclipse/bundles/org.apache.syncope.ide.eclipse.tests/pom.xml]: Because to run the tests, surefire needs the plugin module to be already built.

**Comment by tmess567** [comment]
Most of the project is mainly integration with eclipse UI so headless tests wouldn't help much.

**Comment by tmess567** [comment]
The test cases also require eclipse plugin to be built and so should be kept separate from `fit/core-reference` and after `Apache Syncope IDE Eclipse Plugin` module.

**Comment by ilgrosso** [comment]
I'd suggest then to move tests to a dedicated profile in fit/core-reference (as running core is requested), with test dependency of the eclipse plugin.

Tomorrow I can provide more details, see you on IRC

**Comment by ilgrosso** [comment]
You can find [attached](https://github.com/apache/syncope/files/424433/updated34.diff.zip) a diff file I have generated starting from your modifications here.

Basically, I have added the tests module and execution in a dedicated `eclipse-it` profile, **not** active by default: this means that such tests are not ordinarily executed. I have reverted your changes to `travis.yml` for this reason.

Now, after running `mvn -PskipTests,all` from the root directory, you can move to `ide/eclipse/bundles/org.apache.syncope.ide.eclipse.tests` and launch `mvn -P eclipse-it`: this will first start the Syncope core locally via cargo, then execute the available tests.

I have tried on my laptop and it works.

You should now revert your modifications here and manually apply the attached diff file and check if everything is fine.

Thanks!

**Comment by ilgrosso** [comment]
There was some problem with yesterday's patch: 
[this new one](https://github.com/apache/syncope/files/426824/new34.diff.zip) can be applied on top of your existing repo.

**Comment by ilgrosso** [comment]
Just merged, thank you

**Pull request closed without merging**